src/accordian/Accordion.tsx

```diff
@@ -0,0 +1,197 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as React from "react";
+import {classNames} from "..//utils";
+import {ClassNames} from "./ClassNames";
+import {Icon, Direction} from "../icon";
+
+type AccordianProps = {
+    style?: any;
+    className?: string;
+    content: accordionContent[];
+    accordionMultiPanel?: boolean;
+};
+
+type accordionContent = {
+    title: string;
+    itemComponent: React.ReactNode;
+};
+
+type SignPostState = {
+    panelItems: any;
+    prevItemIndex: number;
+};
+
+export class Accordion extends React.Component<AccordianProps> {
+    state: SignPostState = {
+        prevItemIndex: -1,
+        panelItems: [],
+    };
+
+    componentDidMount() {
+        this.getAccordionContent();
+    }
+    handleButtonClick = (index: any, accordionMultiPanel: any) => {
+        let items = this.state.panelItems;

```

Comment:
It would be a little cleaner to add `const {prevItemIndex} = this.state` here to avoid all of the `this.state.prevItemIndex` being used.

Reply:
Addressed